src/Arcus.Observability.Tests.Unit/Telemetry/ILoggerExtensionsTests.cs

```diff
@@ -0,0 +1,274 @@
+﻿using System;
+using System.Globalization;
+using System.Net;
+using System.Net.Http;
+using Arcus.Observability.Telemetry.Core;
+using Bogus;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Arcus.Observability.Tests.Unit.Telemetry
+{
+    [Trait("Category", "Unit")]
+    public class ILoggerExtensionsTests
+    {
+        private readonly Faker _bogusGenerator = new Faker();
+
+        [Fact]
+        public void LogMetric_ValidArguments_Succeeds()
+        {
+            // Arrange
+            var logger = new TestLogger();
+            string metricName = _bogusGenerator.Name.FullName();
+            double metricValue = _bogusGenerator.Random.Double();
+
+            // Act
+            logger.LogMetric(metricName, metricValue);
+
+            // Assert
+            var logMessage = logger.WrittenMessage;
+            Assert.StartsWith(MessagePrefixes.Metric, logMessage);
+            Assert.Contains(metricName, logMessage);
+            Assert.Contains(metricValue.ToString(CultureInfo.InvariantCulture), logMessage);
+        }
+
+        [Fact]
+        public void LogMetric_NoMetricNameWasSpecified_ThrowsException()
+        {
+            // Arrange
+            var logger = new TestLogger();
+            string metricName = null;
+            double metricValue = _bogusGenerator.Random.Double();
+
+            // Act & Arrange
+            Assert.Throws<ArgumentException>(() => logger.LogMetric(metricName, metricValue));
+        }
+
+        [Fact]
+        public void LogEvent_ValidArguments_Succeeds()
+        {
+            // Arrange
+            var logger = new TestLogger();
+            string eventName = _bogusGenerator.Name.FullName();
+
+            // Act
+            logger.LogEvent(eventName);
+
+            // Assert
+            var logMessage = logger.WrittenMessage;
+            Assert.StartsWith(MessagePrefixes.Event, logMessage);
+            Assert.Contains(eventName, logMessage);
+        }
+
+        [Fact]
+        public void LogEvent_NoEventNameSpecified_ThrowsException()
+        {
+            // Arrange
+            var logger = new TestLogger();
+            string eventName = null;
+
+            // Act & Arrange
+            Assert.Throws<ArgumentException>(() => logger.LogEvent(eventName));
+        }
+
+        [Fact]
+        public void LogSqlDependency_ValidArguments_Succeeds()
```

Comment:
Nicely tested, easy read and correct assertion. Cool! 🥇

Reply:
I learned from the best!